sub-packages/bionemo-esm2/src/bionemo/esm2/testing/compare.py

```diff
@@ -53,34 +51,14 @@ def assert_model_equivalence(
         "MKTVRQERLKSIVRILERSKEPVSGAQLAEELSVSRQVIVQDIAYLRSLGYNIVATPRGYVLA",
         "MKTVRQERLKSI<mask>RILERSKEPVSGAQLAEELS<mask>SRQVIVQDIAYLRSLGYN<mask>VATPRGYVLAGG",
     ]
-    tokens = tokenizer(test_proteins, return_tensors="pt", padding=True, truncation=True).to("cuda")
-    input_ids = tokens["input_ids"]
-    attention_mask = tokens["attention_mask"]
-
-    dtype = get_autocast_dtype(precision)
-    nemo_config = ESM2Config(
-        initial_ckpt_path=str(ckpt_path),
-        include_embeddings=True,
-        include_hiddens=True,
-        params_dtype=dtype,
-        pipeline_dtype=dtype,
-        autocast_dtype=dtype,
-        bf16=dtype is torch.bfloat16,
-        fp16=dtype is torch.float16,
-    )
-
-    nemo_model = nemo_config.configure_model(tokenizer).to("cuda").eval()
 
-    if dtype is torch.float16 or dtype is torch.bfloat16:
-        nemo_model = Float16Module(nemo_config, nemo_model)
-
-    nemo_output = nemo_model(input_ids, attention_mask)
+    nemo_output = eval_esm2(ckpt_path, test_proteins, precision=precision)
```

Comment:
IMO I like to ensure the tensor inputs to the model are the same, rather than doing that twice here. Might make sense to decompose your eval_esm2 function into more modular pieces so that's possible

Reply:
I figured we might run into cases where we iterate on a dataloader, so I separated the setup/teardown to avoid configuring the model every time we call forward.

Now I can also add a method to check the input to eval and tokenize if they are sequences. But I don't think that would look clean